### ISSUE TITLE
Not installing latest Pulumi CLI

### ISSUE THREAD

**Issue by t0yv0** [pulumi/actions#933]
### What happened?

https://github.com/pulumi/pulumi-terraform-bridge/actions/runs/4882937968/jobs/8713629269#step:6:1

### Expected Behavior

Expected the action to install latest Pulumi CLI

### Steps to reproduce

Instead it installed an older version

### Output of `pulumi about`

N/A

### Additional context

N/A

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by RobbieMcKinstry** [comment]
Ahh... I think that's "working as expected!" :(
I read "^3" as "version 3 or higher". I think you're reading it as "the highest version less than 4". 

With my reading, v3.64.0 satisfies the constraint and comes pre-downloaded on the machine, so for the sake of performance, we use the cached version.

WDYT?

**Comment by t0yv0** [comment]
Yes it makes sense. Is there an option to check/force latest? For my use case I can workaround to not rely on this at all .

**Comment by RobbieMcKinstry** [comment]
Hm, I don't think you can, because I think our Semver library doesn't recognize `latest`.

**Comment by t0yv0** [comment]
Got it. Well thanks for checking, can close as won't fix.

**Comment by RobbieMcKinstry** [comment]
FWIW we're looking to rewrite how we grab the list of versions and decide which to use. See https://github.com/pulumi/actions/issues/916 for more info. At that time, we could reasonably add "latest" support too. There might even be a way to do this programmatically from the GitHub API. Maybe we can query the `Releases` API for the "latest" release.

**Comment by RobbieMcKinstry** [comment]
Boing! https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-the-latest-release
